js/script.js

```diff
@@ -31,91 +38,171 @@ function resizeHandler() {
+    } else {
+        css = {
+            opacity: 1,
+            position: 'fixed',
+            top: 0,
+            transition: 'top 0.4s'
+        };
     }
+    header.css(css);
+    afterScroll();
+}
 
-    var scrollTimeout;
-    var lastScrollTop = 0;
-    var windowHeight = $(window).height();
-
-    $(window).scroll(function () {
-        if (scrollTimeout) {
-            clearTimeout(scrollTimeout);
-            scrollTimeout = null;
-        }
-        scrollTimeout = setTimeout(scrollHandler, 150);
+function handleScrollingUp(current) {
+    var sections = [];
+    section.each(function() {
+        var sectionInfo = {
+            position: $(this).position().top,
+            hash: $(this).attr('data-section-name')
+        };
+        sections.push(sectionInfo);
     });
 
-    scrollHandler = function() {
-        var scrollTop = $(this).scrollTop();
-        if ($(window).width() >= 375) {
-            handleScrollifyScrolling(lastScrollTop, scrollTop);
-        } else {
-            handleNormalScrolling(lastScrollTop, scrollTop);
+    for (var index in sections) {
+        if (current < sections[index].position) {
+            window.location.hash = sections[index].hash;
+            break;
         }
-        lastScrollTop = scrollTop;
     }
+    $.scrollify.update();
+    disableScrollify();
+}
+
+function handleScrollingDown(current) {
+    if (!isScrolling) {
+        handleScrollifyTransition(current);
+    }
+}
 
-    handleNormalScrolling = function(lastScrollTop, scrollTop) {
-        stickyHeaderOnScroll(scrollTop, lastScrollTop);
+function handleScrollifyTransition(current) {
+    var overFlow = $.scrollify.current().height() - windowHeight;
+    if (current > $.scrollify.current().position().top + overFlow) {
+        isScrolling = true;
+        scrollToNext();
     }
+}
 
-    handleScrollifyScrolling = function(lastScrollTop, scrollTop) {
-        if (scrollTop === 0 || scrollTop >= lastScrollTop) {
-                if ($.scrollify.isDisabled()) { $.scrollify.enable(); }
-                setTouchAction('none');
-        } else {
-            $.scrollify.disable();
-            setTouchAction('auto');
-        }
-        stickyHeaderOnScroll(scrollTop, windowHeight);
+function scrollToNext() {
+    enableScrollify();
+    $.scrollify.update();
+    $.scrollify.next();
+    disableScrollify();
+    setTimeout(afterScroll, 1250);
+}
+
+function afterScroll() {
+    lastScrollTop = $(window).scrollTop();
+    setTouchAction('auto');
+    isScrolling = false;
+}
+
+function enableScrollify() {
+    if ($.scrollify.isDisabled()) {
+        $.scrollify.enable();
     }
+}
 
-    stickyHeaderOnScroll = function(scrollTop, point) {
-        if ($.scrollify.isDisabled() || (scrollTop < point)) {
-            setStickyHeaderTween(0, 1);
-        } else {
-            setStickyHeaderTween('-130px', 0);
-        }
+function disableScrollify() {
+    if (!$.scrollify.isDisabled()) {
+        $.scrollify.disable();
     }
```

Comment:
Tilan tarkistaminen ja sitten tilan asettaminen on hitaampaa kuin pelkkä tilan asettaminen. Olettaisin että tässä koodissa mennään yleensä "happy path" periaatteella, eli scrollausta ei disabloida jollei tiedetä että se on enabloitu ja toisinpäin. Niinpä asian iffittäminen hidastaa "99%" kutsuista. En välttämättä tekisi eri funktioita näin yksinkertaiselle kutsulle kuin enable() ja disable().